documentation/general/torn-sdk.md

```diff
@@ -0,0 +1,142 @@
+
+## Visual Studio using Visual Studio Analyzers
+
+### .NET SDK in box analyzers dual insert
+
+Analyzers which ship in the .NET SDK box will change to having a copy checked into Visual Studio. When .NET SDK based projects are loaded at design time, the Visual Studio copy of the analyzer will be loaded. Roslyn already understands how to prefer Visual Studio copies of analyzers. That work will need to be extended a bit but that is pretty straight forward code.
+
+This approach enables us to take actions like NGEN or R2R analyzers inside of Visual Studio. This is a long standing request from the Visual Studio perf team but very hard to satisfy in our current
+
+This approach is already taken by [the Razor generator][code-razor-vs-load]. This work was done for other reliability reasons but turned out working well for this scenario. There is initial upfront work to get the Visual Studio copy loading in design time builds but it has virtually zero maintanence cost.
+
+This also means these analyzers can vastly simplify their development model by always targeting the latest version of Roslyn. There is no more need to multi-target because the version of the compiler the analyzer will be used with is known at ship time for all core build scenarios.
+
+This does mean that our design time experience can differ from our command line experience when customers are in a torn state. Specifically it's possible, even likely in some cases, that the diagonstics produced by design time builds will differ from command line builds. That is a trade off that we are willing to make for reliability. Customers who wish to have a consistent experience between design time should not operate in a torn state.
+
+## .NET SDK in box analyzers target oldest
+
+Analyzers which have no need to target the latest Roslyn could instead choose to target the oldest supported version of Roslyn. That ensures they can safely load into any supported scenario without issue.
+
+This strategy should be considered short term. The Roslyn API is under constant development to respond to the performance demands of Visual Studio. New APIs can quickly become virtually mandatory for analyzers and generators to use and will only be available on latest. At that point the analyzer will need to also update to use dual insertions.
+
+### NuGet based analyzers
+
+Analyzers which ship via NuGet will continue to following the existing [support policies][matrix-of-paine]. This means that they will either need to target a sufficiently old version of Roslyn or implement multi-targeting support.
```

Comment:
How many third party analyzers are there and do we have a forum for giving them a heads up on our changes here (and are they even aware of the matrix)?

Reply:
There are lots of 3rd party analyzers out there. Probably the best forum is announcements on the dotnet blog or announcement issues on the roslyn repo. 

At the same time, I'm not sure what we would announce there. Any NuGet based analyzer (3rd party or 1rst party) is subject to being run under multiple compiler versions as they can be installed in any .NET SDK. Analyzer multi-targeting is already done based on roslyn version, not SDK version. So that means there is really no change here at all.

Yes in a torn state they're likely to be run on a more recent compiler. That is the same compiler they'd be run on in a `dotnet  build` state so it's not a new scenario. 

I can overall expand on this a bit to make it clearer why this is not a meaningful change for them.